Arm/Insts/Common.lean

```diff
@@ -144,6 +144,18 @@ theorem CheckSPAlignment_AddWithCarry_64_4 (st : ArmState) (y : BitVec 64) (carr
   simp_all only [CheckSPAlignment, read_gpr, zeroExtend_eq, Nat.sub_zero, add_eq,
     Aligned_AddWithCarry_64_4]
 
+theorem CheckSPAlignment_of_r_sp_eq {s s' : ArmState}
```

Comment:
Needs a `state_simp_rules` tag? Also for the theorem below?

Reply:
Done